### PR TITLE
AGP 8.

### PR DESCRIPTION
**Pull request by zhaobozhen**

### CODE REVIEW

**Comment by zhaobozhen** [comment]
```
11 problems were found storing the configuration cache, 9 of which seem unique.
- Build file 'app/build.gradle.kts': external process started 'git rev-list --count HEAD'
  See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:external_processes
- Build file 'app/build.gradle.kts': external process started 'git rev-parse --short HEAD'
  See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:external_processes
- Build file 'app/build.gradle.kts': external process started 'git tag -l 2.4.0'
  See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:external_processes
- Task `:app:generateFossReleaseProto` of type `com.google.protobuf.gradle.GenerateProtoTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
- Task `:app:generateMarketReleaseProto` of type `com.google.protobuf.gradle.GenerateProtoTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
- Task `:app:optimizeFossReleaseResources` of type `com.android.build.gradle.internal.tasks.OptimizeResourcesTask`: cannot serialize Gradle script object references as these are not supported with the configuration cache.
  See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:app:optimizeFossReleaseResources` of type `com.android.build.gradle.internal.tasks.OptimizeResourcesTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
- Task `:app:optimizeMarketReleaseResources` of type `com.android.build.gradle.internal.tasks.OptimizeResourcesTask`: cannot serialize Gradle script object references as these are not supported with the configuration cache.
  See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:app:optimizeMarketReleaseResources` of type `com.android.build.gradle.internal.tasks.OptimizeResourcesTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
```

**Comment by zhaobozhen** [comment]
```
2023-04-02T03:26:00.3287792Z 7 problems were found storing the configuration cache, 5 of which seem unique.
2023-04-02T03:26:00.3289170Z - Build file 'app/build.gradle.kts': external process started 'git rev-list --count HEAD'
2023-04-02T03:26:00.3303541Z   See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:external_processes
2023-04-02T03:26:00.3304496Z - Build file 'app/build.gradle.kts': external process started 'git rev-parse --short HEAD'
2023-04-02T03:26:00.3305491Z   See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:external_processes
2023-04-02T03:26:00.3306544Z - Build file 'app/build.gradle.kts': external process started 'git tag -l 2.4.0'
2023-04-02T03:26:00.3307379Z   See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:external_processes
2023-04-02T03:26:00.3308349Z - Task `:app:generateFossReleaseProto` of type `com.google.protobuf.gradle.GenerateProtoTask`: invocation of 'Task.project' at execution time is unsupported.
2023-04-02T03:26:00.3312049Z   See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
2023-04-02T03:26:00.3313162Z - Task `:app:generateMarketReleaseProto` of type `com.google.protobuf.gradle.GenerateProtoTask`: invocation of 'Task.project' at execution time is unsupported.
2023-04-02T03:26:00.3314129Z   See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
2023-04-02T03:26:00.3314605Z
```

**Comment by zhaobozhen** [comment]
#https://github.com/google/protobuf-gradle-plugin/issues/687

**Comment by zhaobozhen** [comment]
@Goooler Any idea to workaround this?

```
2023-04-02T03:26:00.3289170Z - Build file 'app/build.gradle.kts': external process started 'git rev-list --count HEAD'
2023-04-02T03:26:00.3303541Z   See https://docs.gradle.org/8.1-rc-2/userguide/configuration_cache.html#config_cache:requirements:external_processes

```

**Comment by Goooler** [comment]
Need to rename the CC flag.

https://docs.gradle.org/8.1/userguide/upgrading_version_8.html#configuration_caching_options_renamed

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -57,8 +57,9 @@ jobs:
           fetch-depth: 0
       - uses: actions/setup-java@v3
         with:
-          distribution: 'zulu'
+          distribution: 'temurin'
           java-version: 17
+          cache: gradle
       - uses: gradle/gradle-build-action@v2
```
Comment (Goooler) [inline .github/workflows/android.yml]: Why need this change? 
- Gradle is cached by `gradle/gradle-build-action@v2`.
- It's better to use `zulu`.

**Review by Goooler — APPROVED** [review]

**Pull request merged**